src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/ReleaseInfoService.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Publisher.Model;
+using GovUk.Education.ExploreEducationStatistics.Publisher.Services.Interfaces;
+using Microsoft.Azure.Cosmos.Table;
+
+namespace GovUk.Education.ExploreEducationStatistics.Publisher.Services
+{
+    public class ReleaseInfoService : IReleaseInfoService
+    {
+        private readonly ITableStorageService _tableStorageService;
+
+        private const string TableName = "releases";
+
+        public ReleaseInfoService(ITableStorageService tableStorageService)
+        {
+            _tableStorageService = tableStorageService;
+        }
+
+        public async Task AddReleaseInfoAsync(QueueReleaseMessage message, ReleaseInfoStatus status)
+        {
+            var releaseInfoTable = await GetReleaseInfoTableAsync();
+            var releaseInfo = new ReleaseInfo(message.PublicationSlug,
+                message.PublishScheduled,
+                message.ReleaseId,
+                message.ReleaseSlug,
+                status);
+            await releaseInfoTable.ExecuteAsync(TableOperation.InsertOrReplace(releaseInfo));
+        }
+
+        public async Task<IEnumerable<ReleaseInfo>> GetScheduledReleasesAsync()
+        {
+            var results = new List<ReleaseInfo>();
+
+            var releaseInfoTable = await GetReleaseInfoTableAsync();
+            TableContinuationToken token = null;
+            do
+            {
+                // TODO EES-863 Currently returns all scheduled releases
+                // TODO EES-863 Only query scheduled releases that are being published today
+                var tableQuery = new TableQuery<ReleaseInfo>()
+                    .Where(TableQuery.GenerateFilterCondition("Status",
+                        QueryComparisons.Equal, ReleaseInfoStatus.Scheduled.ToString()));
+
+                var queryResult = await releaseInfoTable.ExecuteQuerySegmentedAsync(tableQuery, token);
+
+                results.AddRange(queryResult.Results);
+                token = queryResult.ContinuationToken;
+            } while (token != null);
+
+            return results;
+        }
+
+        public async Task UpdateReleaseInfoStage(Guid releaseId, Guid releaseInfoId)
+        {
+            var releaseInfoTable = await GetReleaseInfoTableAsync();
+            var tableResult = await releaseInfoTable.ExecuteAsync(
+                TableOperation.Retrieve<ReleaseInfo>(releaseId.ToString(), releaseInfoId.ToString()));
+
+            if (tableResult.Result is ReleaseInfo releaseInfo)
+            {
+                releaseInfo.ReleaseContentStage = 999;
```

Comment:
What's this in aid of?

Reply:
It came from starting the work on EES-863 to add statuses about individual tasks that might be in progress/failed/complete. I've improved this now to make it more complete.